### ISSUE TITLE
Error column doesn't exist when  creating view 

### ISSUE THREAD

**Issue by JeeIPI** [darold/ora2pg#526]
ora2Pg: 18.2 

Issue description:
--------------------

-  In Oracle database create one view  as follow:
   create view  GILLED  as SELECT * FROM DUAL;
   Doing this, the SELECT  SQL statement corresponding to the view generated by Oracle is 
       ->       SELECT "DUMMY" FROM DUAL;

- So ora2pg  generates  the following  create view SQL statement:

    CREATE OR REPLACE FORCE VIEW GILLED
          (    DUMMY)
           BEQUEATH DEFINER
     AS
         SELECT "DUMMY" FROM DUAL;

   which fails  with "column doesn't exist" error as   "DUMMY"  column doesn't exist in Postgres database. Postgres knows  "dummy"  or DUMMY or dummy  but doesn't know "DUMMY".

Possible solutions:
--------------------

 1 - Drop the double quote enclosing :  "DUMMY"  becomes  DUMMY
    ( Doesn't work if some SQL reserved keywords are used as column_name. ex : column name "PLAN" )

or

2 - Set the column name in lowercase:  "DUMMY" becomes "dummy"   as Postgres stores internal meta-data in lowercase. 
  ( Should work in all the cases )

Many thanks for your help.
Best regards,
Jean-Pierre

**Comment by darold** [comment]
Are you using the `-p` option or have you enable the PLSQL_PGSQL configuration directive? Here it seems that there is no translation at all.

**Comment by JeeIPI** [comment]
Having a look in the shell script launching ora2pg, I found the following:
 ora2pg -p -t VIEW -o views.sql -b ${REP_DMP}/schema/views -c ${PO_FIC_CONFIG}

So, I think option -p is active. 

Thanks a lot for your help.
Best Regards,
Jean-Pierre

**Comment by darold** [comment]
Ok, I understand. When I create this view on an Oracle database, the translation obtained is the following:
```
CREATE OR REPLACE VIEW dummy_view (dummy) AS SELECT "DUMMY";
```
Which is a wrong translation for this special case, an expected translation could be:
```
CREATE OR REPLACE VIEW dummy_view (dummy) AS SELECT 1 AS DUMMY;
```
This is a very special case and it is perhaps more appropriate to replace the call to the dummy view by a simple call to `SELECT 1` in your code.

**Comment by jpeep44** [comment]
Let me try to clarify the issue.

1 -  Imagine you have a table named T1 with 2 columns C1 and C2  in an Oracle database.
2 - Always in Oracle database, you create a view on this table with the following SQL statement:
      `create view V1 as select * from T1;`
3 - Instead of storing  in his dictionary  `select * from T1`, Oracle stores  `select "C1", "C2" from T1`  
4 - Moving the view V1 from Oracle to PostgreSQL using ora2pg fails because `select "C1", "C2" from T1` is incorrect for a PostgreSQL database.  The correct PostgreSQL SQL statement to create this view is `select "c1", "c2" from T1`. For PostgreSQL, the column name enclosed by double quote must be in lowercase.

I hope my explanations are better this time. Of course, feel free to ask more details if you need.

Many thanks for your great work.
Best Regards,
Jean-Pierre

**Comment by darold** [comment]
This is not the default behavior of Ora2Pg, I think you have enabled PRESERVE_CASE in the ora2pg.conf, this is why the column's names are enclosed between double quote.

**Comment by jpeep44** [comment]
I will check tomorrow morning and I will let you know. As this is someone else than me who did the ora2pg, I have to check this point.

Thanks again for your great help.
Best Regards,
Jean-Pierre

**Comment by JeeIPI** [comment]
I checked the ora2pg configuration.  
I found only one ora2pg.conf on the server. It  contains the following:
```
...
# By default all object names are converted to lower case, if you
# want to preserve Oracle object name as-is set this to 1. Not recommended
# unless you always quote all tables and columns on all your scripts.
PRESERVE_CASE   0
...
```
My understanding is with this setup, ora2pg should convert the column_name from uppercase to lowercase but for any reason doesn't do it.
Maybe, this is because the view column_name are already enclosed by double quote in Oracle dictionnary ...

Many thanks for your help.
Best Regards,
Jean-Pierre

**Comment by darold** [comment]
Fixed in commit f43ddfe.

**Comment by jpeep44** [comment]
Many thanks. 
Best Regards,
Jean-Pierre
